### ISSUE TITLE
Increasing Tacotron teacher forcing start delay can help quick alignment

### ISSUE THREAD

**Issue by begeekmyfriend** [Rayhane-mamah/Tacotron-2#284]
Generally speaking, with the latest version and the default dataset the training would learn to alignment after 30K steps which means this project has better generalization ability at the cost of pricision loss in my humble opinion. The evaluation effect is good but it takes a bit longer for training. 

Fortunately I find that the soft teacher forcing in Tacotron training helper might take the responsiblity for this issue. I have tried porting `_teacher_forcing_ratio_decay` method into my modified [T1](https://github.com/begeemyfriend/tacotron) project. And what I found is that the global step point of alignment during training was obviously lagged. In general the alignment step point is about 10~20K, but once the soft teacher forcing added, the alignment occurs after 40K steps. Therefore I doubt there should have been more tearcher forcing steps. However when I increased the `tacotron_teacher_forcing_start_decay` hyper parameter into 20K, the alignment occurred before 20K steps. So I think I have verified that the teacher forcing is helpful to the alignment with my own experiments.

I suggest we need to adjust the hyper parameter `tacotron_teacher_forcing_start_decay` to make sure that **the soft teacher forcing ratio start to change after the alignment is learned**. In my experience, 20K is recommended.

By the way, T2 uses more regularization items than T1 such as L2, dropout in convolution and zoneout LSTM which are also take the responsibility for lagging of alignment but with the benefit of preventing  overfitting for better generization. The T2 evaluation is better than that of T1.
![step-16000-align](https://user-images.githubusercontent.com/6031938/49272895-1b0b1c80-f4ae-11e8-8f8b-5d6b15ccbdbe.png)
![step-45000-align](https://user-images.githubusercontent.com/6031938/49272982-63c2d580-f4ae-11e8-8bae-fce2987b3e4e.png)

**Comment by begeekmyfriend** [comment]
By the way, I think the soft teacher forciing ratio is really a great improvement for overfitting. I have got better results for male corpus evaluation in T1 model. Well done @Rayhane-mamah

**Comment by Rayhane-mamah** [comment]
@begeekmyfriend, hi tganks for the status update!

I assume you're talking about scheduled teacher forcing (Curriculum Learning) right?
I am aware that, depending on the data nature, it helps T2 generalize with less data as it gradually moves the model distribution from train to test mode.

I agree with your 20k start decay, I forgot to update it as default branch uses constant learning (

**Comment by begeekmyfriend** [comment]
What is more, in my point of view, clipping in normalization is unnecessary because it might lose some edge information. When I cancel that clipping it still works well for both of alignment and evaluation. What do you think of that?
